web/src/app/services/ServiceMaintenanceDialog.tsx

```diff
@@ -25,19 +26,35 @@ function label(hours: number): JSX.Element {
 }
 
 function ServiceMaintenanceForm(props: {
-  onChange: (val: number) => void
-  selectedIndex: number
+  value: DateTime
+  onChange: (val: DateTime) => void
 }): JSX.Element {
+  const [selectedOption, setSelectedOption] = useState(1)
+
+  function handleOption(value: number): void {
+    setSelectedOption(value)
+    if (value === -1) return
+    props.onChange(DateTime.local().plus({ hours: value }))
+  }
+
   return (
     <FormControl>
       <RadioGroup
-        value={props.selectedIndex}
-        onChange={(e) => props.onChange(parseInt(e.target.value, 10))}
+        value={selectedOption}
+        onChange={(e) => handleOption(parseInt(e.target.value, 10))}
       >
         <FormControlLabel value={1} control={<Radio />} label={label(1)} />
         <FormControlLabel value={2} control={<Radio />} label={label(2)} />
         <FormControlLabel value={4} control={<Radio />} label={label(4)} />
+        <FormControlLabel value={-1} control={<Radio />} label='Specify' />
       </RadioGroup>
+      <ISODateTimePicker
```

Comment:
I think some left and top padding would help here. Left padding to align with the radio buttons and top padding just to nudge it down slightly.